src/TheThingsNetwork.cpp

```diff
@@ -396,19 +406,45 @@ void TheThingsNetwork::autoBaud()
   while (attempts-- && length == 0)
   {
     delay(100);
+    // Send break + Autobaud 
     modemStream->write((byte)0x00);
     modemStream->write(0x55);
     modemStream->write(SEND_MSG);
-    sendCommand(SYS_TABLE, 0, true, false);
-    sendCommand(SYS_TABLE, SYS_GET, true, false);
-    sendCommand(SYS_TABLE, SYS_GET_VER, false, false);
-    modemStream->write(SEND_MSG);
-    length = modemStream->readBytesUntil('\n', buffer, sizeof(buffer));
+    // check we can talk
+    length = checkComm();
+    
+    // We succeded talking to the module ?
+    baudDetermined = length ? true : false;
   }
   delay(100);
   clearReadBuffer();
   modemStream->setTimeout(10000);
-  baudDetermined = true;
+}
+
+void TheThingsNetwork::wake()
+{  
+  HardwareSerial *pLora = (HardwareSerial*) modemStream;
```

Comment:
We can't do this; it's not guaranteed that `modemStream` is a `HardwareSerial`. In fact, we use quite some alternatives

Reply:
The library is called TTN Arduino device lib, and TTN devices (The Things Node and The Things Uno) are for now connected and using hardware serial of the 32U4.
I think focus should be fixing Low Power issues (and this fix works) until we have another device using another method than hardware Serial. In this case we could resolve with some `#define` and boards.txt definition)
Anyway I'm opened to any other alternatives. Just let me know

Reply:
This library is intended to use with any Microchip RN2xx3 module, whether or not on The Things Uno or The Things Node, for example SODAQ boards or Arduinos with the RN module on a shield. Therefore, we cannot introduce a dependency on specific serials, let alone a cast like this, unfortunately.

Reply:
Ok, I've implemented the same HAL than in Sodaq_RN2483 library so it use 

```cpp
#if defined(ARDUINO_ARCH_AVR)
typedef HardwareSerial SerialType;
#elif defined(ARDUINO_ARCH_SAM) || defined(ARDUINO_ARCH_SAMD)
typedef Uart SerialType;
#else
typedef Stream SerialType;
#endif
```
and of course changing Stream to SerialType in the library, should do the trick.